### DISCUSSION TITLE
Different permissions depending on rows accessed

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#4518] [Feature Requests]
A nice addition to the rule based permission system is to have different permissions for different sets of fields for the same user.

For example, an author can create an article. The author can read all fields in the article they created.
However, the same author shouldn't be able to read all those fields on articles they didn't create. This is currently not possible to do.

One way to do it is to allow multi roles per user, allow role switching, or allow mirroring views (since different permissions can be defined on different views for the same collection)

**Comment by rijkvanzanten** [comment]
This ties in nicely with https://github.com/directus/directus/discussions/3950 as well

**Reply by ghost** [reply]
Yes! That would make the example use case above possible with something like:
```
scope: user_created _eq $CURRENT_USER
validation: (can read all rows)

scope: *
validation: (can read only specific public fields)
```
